protocol/v2/qbft/controller/controller.go

```diff
@@ -123,7 +122,7 @@ func (c *Controller) ProcessMsg(logger *zap.Logger, signedMessage *spectypes.Sig
 }
 
 func (c *Controller) UponExistingInstanceMsg(logger *zap.Logger, msg *specqbft.ProcessingMessage) (*spectypes.SignedSSVMessage, error) {
-	inst := c.InstanceForHeight(logger, msg.QBFTMessage.Height)
+	inst := c.StoredInstances.FindInstance(msg.QBFTMessage.Height)
```

Comment:
@nkryuchkov i think that since if we were still saving historical instances, it would make sense to still grab them here for full nodes, but since we don't anymore then we can probably do nothing but fail here

this is likely fine though because full nodes rarely restart, so recent instances should still be in-memory, and in case of restarts, it takes some time to find peers anyway so historical instances become irrelevant for incoming messages very fast

Reply:
> most historical instances from last restart likely became irrelevant anyway (remember that incoming messages must refer to last 2 slots or so according to msg validation rule)

@moshe-blox I happened to pass by and check, wouldn't attestation duties still need to be performed even after restart ? Seems like they have [all 32 slots to them](https://github.com/ssvlabs/ssv/blob/8d18d2fd34181130e4dbfd2a6ffe4c053b174831/message/validation/common_checks.go#L36-L52) from message-validation standpoint, it's probably long to recover from SSV node restart (the only thing is - round 7,8 timeouts are 120s which might be a bit too long specifically for node restart scenario - wouldn't it be better to have more rounds with shorter timeouts, say 20s each) ?

Reply:
@iurii-ssv im not sure that it was ever possible to recover after restart and join a late round, i dont think we ever tested this

my point was that even if it's possible, it's likely not worth the effort, because an attestation that has not submitted during its first few slots is already most likely a lost cause (the operators are likely down, and late submissions are less likely to succeed), and when combining with the fact that restarts are quite rare then the potential gain here seems very tiny

the reason for long timeouts past round 8 is to be like any good retry mechanism: to increase retry interval after failures and avoid spamming